lib/Admin.php

```diff
@@ -582,6 +582,16 @@ public function add_image_to_library(): void {
 			}
 		}
 
+		if ( 'generic' === $asset_data['type'] ) {
```

Comment:
One suggestion that may help with makings this easy to maintain is to create a constants file with these strings. In the case of 'generic', I see that it is used at least 3 times just in this PR alone. With it as a constant, if the value ever needed to be changed, we could do this in one location and have it updated throughout the system.